docs/madr/decisions/057-rework-mesh-routes-matching.md

```diff
@@ -0,0 +1,897 @@
+# Rework of Mesh*Routes matching
+
+* Status: accepted
+
+Technical story: https://github.com/kumahq/kuma/issues/10247
+
+## Context and Problem Statement
+
+At the moment when targeting `Mesh*Route` you need to put it in topLevel targetRef. 
+We want to iterate over policy design and move `Mesh*Route` to `spec.to[].targetRef` section follow model where `spec.targetRef`
+selects whole proxies, `spec.from[].targetRef` selects inbounds and `spec.to[].targetRef` selects outbounds
+
+## Decision Outcome
+
+- add possibility to put `Mesh*Routes` in `spec.to[].targetRef`
+
+## Considered options
+
+1. add possibility to put `Mesh*Routes` in `spec.to[].targetRef` -> `spec.targetRef` selects proxies, `spec.from[].targetRef` selects inbounds, `spec.to[].targetRef` selects outbounds.
+2. leave `Mesh*Routes` in top level targetRef as it is now, improve the semantics of `spec.targetRef` and introduce `spec.workloadSelector` to select whole proxies
+
+### Add possibility to put Mesh*Routes in `spec.to[].targetRef`
+
+#### Advantages
+
+- simplifying policy api as `spec.targetRef` will always pick proxy and `spec.to[].targetRef` section will select outgoing traffic
```

Comment:
> `spec.to[].targetRef` section will select outgoing traffic

This isn't really the whole picture. The `to` section would also pick the proxy.

Current:
- `spec.targetRef` picks both the proxy and the outgoing traffic
- `spec.to[].targetRef` is always `Mesh`

Proposal:
- `spec.targetRef` picks the proxy
- `spec.to[].targetRef` picks both the proxy and the outbound

Reply:
removed the point